superset/assets/javascripts/explore/components/ExploreViewContainer.jsx

```diff
@@ -24,14 +25,76 @@ const propTypes = {
 };
 
 class ExploreViewContainer extends React.Component {
+
   constructor(props) {
     super(props);
     this.state = {
       height: this.getHeight(),
       showModal: false,
+      table: Array(0),
+      data: [],
     };
   }
 
+  statsPairedTTest(aa, bb) {
+      // calculate the paired t-test values between two arrays
+      var diff, ii, sum = 0, nn = 0, ss = 0, tvalue;
+      var dist = require('distributions');
+      for (ii = 0; ii < aa.length; ii += 1) {
+        diff = bb[ii]['y'] - aa[ii]['y'];
+        if (global.isFinite(diff)) {
+          nn += 1;
+          sum += diff;
+          ss += diff * diff;
+        }
+      }
+      tvalue = -Math.abs(sum * Math.sqrt((nn - 1) / (nn * ss - sum * sum)));
+      try {
+        return (2 * new dist.Studentt(nn - 1).cdf(tvalue)).toFixed(6);
+      } catch (error) {
+        return NaN;
+      }
+  }
+
+  statsLift(aa, bb) {
+    // compute the lift between two arrays
+    var a_sum = 0, b_sum = 0;
+    for (var i = 0; i < aa.length; ++i) {
+      a_sum += aa[i]['y'];
+      b_sum += bb[i]['y'];
+    }
+    return ((a_sum/b_sum)*100).toFixed(4);
+  }
+
+  computePairedTTest(control) {
+    // compute the paired t-test values and update state
+    const data = this.state.data.slice();
+    const pValueTable = Array(data.length);
+    for (var i = 0; i < data.length; i += 1) {
+      if (i === control) {
+        pValueTable[i] = { pval: 'control', lift: 'control', stream: data[i].key, highlight: true,};
+        continue;
+      }
+      const currentPairedTTest = this.statsPairedTTest(data[i].values, data[control].values);
+      const currentLift = this.statsLift(data[i].values, data[control].values);
+      pValueTable[i] = { pval: currentPairedTTest, lift: currentLift, stream: data[i].key, highlight: false,};
+    }
+    this.setState({ table: pValueTable });
+  }
+
+  pairedTTestQuery() {
+    // check the viz_type, grab the required data, update state and compute the paired t-test values
+    if (this.props.form_data.viz_type === 'line_ttest') {
+      const url = getExploreUrl(this.props.form_data, 'json');
+      return $.getJSON(url)
+        .then((results) => {
+          const data = results.data;
+          this.setState({ data: data });
+          this.computePairedTTest(0);
```

Comment:
0 is just the default index. The user can select any stream to be the control by clicking on the stream's row in the table.